lib/rucio/core/replica.py

```diff
@@ -3253,9 +3589,33 @@ def bulk_add_bad_replicas(replicas, account, state=BadFilesStatus.TEMPORARY_UNAV
     for replica in replicas:
         insert_new_row = True
         if state == BadFilesStatus.TEMPORARY_UNAVAILABLE:
-            query = session.query(models.BadReplicas).filter_by(scope=replica['scope'], name=replica['name'], rse_id=replica['rse_id'], state=state)
-            if query.count():
-                query.update({'state': BadFilesStatus.TEMPORARY_UNAVAILABLE, 'updated_at': datetime.utcnow(), 'account': account, 'reason': reason, 'expires_at': expires_at}, synchronize_session=False)
+            stmt = select(
+                models.BadReplicas
+            ).where(
+                and_(models.BadReplicas.scope == replica['scope'],
+                     models.BadReplicas.name == replica['name'],
+                     models.BadReplicas.rse_id == replica['rse_id'],
+                     models.BadReplicas.state == BadFilesStatus.TEMPORARY_UNAVAILABLE)
```

Comment:
Would be best to use `models.BadReplicas.state == state` here to be consistent with the previous version of the query